pkg/controller/common/defaults/metadata.go

```diff
@@ -4,11 +4,11 @@
 
 package defaults
 
-// SetDefaultLabels append labels from defaults into existing. If a label already exists,
+// AppendMetadata append labels/annotations from defaults into existing. If a label/annotation already exists,
 // its value is not overridden from the one in defaults.
-// One use case here is to inherit user-provided labels, and append our own only if not already
+// One use case here is to inherit user-provided labels/annotations, and append our own only if not already
 // set by the user.
-func SetDefaultLabels(existing map[string]string, defaults map[string]string) map[string]string {
+func AppendMetadata(existing map[string]string, defaults map[string]string) map[string]string {
```

Comment:
`Metadata` feels a bit weird to me since it is a superset of labels + annotations with name, namespace, UID, etc.
Also `Append` feels a bit strange for a map (was there before this PR).

Maybe something like `MergeKeyValue(a, b)`?

Reply:
+1 for `Merge` instead of `Append` but not completely convinced by `KeyValue` which sounds a little too abstract.
Nevermind, while trying to move it to another package, I found:

https://github.com/elastic/cloud-on-k8s/blob/35bc1ba07779dcf5a32eaff54890462caffa6fe2/pkg/utils/maps/maps.go#L38-L54

:D

I'm going to reuse that and delete `AppendMetadata`.